pkg/table/tables/index.go

```diff
@@ -386,6 +386,13 @@ func needPresumeKeyNotExistsFlag(ctx context.Context, txn kv.Transaction, key, t
 	return false, nil
 }
 
+func handleEqual(dupHandle, prevHandle kv.Handle) bool {
```

Comment:
Maybe we could change `func (ph PartitionHandle) Equal(h Handle) bool` function. Only both handles are partition handle, we need compare pid.

Reply:
> Maybe we could change func (ph PartitionHandle) Equal(h Handle) bool function.

I'm not sure if it will affect other code so I add a new function. Let me have a try.